src/main/java/ch/wisv/areafiftylan/utils/mail/MailServiceImpl.java

```diff
@@ -113,11 +114,11 @@ private String prepareHtmlContent(String name, String message) {
     }
 
     private String formatRecipient(User user) {
-        if (user.getProfile() != null) {
-            return user.getProfile().getFirstName() + " " + user.getProfile().getLastName();
-        } else {
-            return user.getEmail();
+        Profile profile = user.getProfile();
+        if (profile != null && profile.getFirstName() != null && profile.getLastName() != null) {
+            return profile.getFirstName() + " " + profile.getLastName();
         }
+        return "User";
```

Comment:
This sounds pretty dev-like. Let's either make this an email address, or turn it into something like

When defined:
>Hi First Last!

When undefined:
>Hi!

Should be easy to implement and avoids the awkward "Hi User"

Reply:
Agreed

Reply:
I prefer the latter (no mail). What would you say?